open_rarity/rarity_ranker.py

```diff
@@ -86,7 +106,10 @@ def set_rarity_ranks(
         """
         sorted_token_rarities: list[TokenRarity] = sorted(
             token_rarities,
-            key=lambda k: k.score,
+            key=lambda k: (
+                k.token_features.unique_attribute_count,
+                k.score,
```

Comment:
might actually be nice to have TokenRarity have a method to return sorted fields.

Reply:
Yeah let's file this as a general improvement of TokenRarity wrapper.  Right now it's just `@dataclass` , let's keep it as simple data object for now.

Reply:
dataclasses can have methods - like how I do for `Token` and `Collection` class